packages/core/contracts/oracle/implementation/AdminIdentifierLib.sol

```diff
@@ -0,0 +1,64 @@
+/**
+ * @title Library to construct admin identifiers.
+ */
+library AdminIdentifierLib {
+    // Returns a UTF-8 identifier representing a particular admin proposal.
+    // The identifier is of the form "Admin n", where n is the proposal id provided.
+    function _constructIdentifier(uint256 id) internal pure returns (bytes32) {
+        bytes32 bytesId = _uintToUtf8(id);
+        return _addPrefix(bytesId, "Admin ", 6);
+    }
+
+    // This method converts the integer `v` into a base-10, UTF-8 representation stored in a `bytes32` type.
+    // If the input cannot be represented by 32 base-10 digits, it returns only the highest 32 digits.
+    // This method is based off of this code: https://ethereum.stackexchange.com/a/6613/47801.
+    function _uintToUtf8(uint256 v) internal pure returns (bytes32) {
```

Comment:
Would this method be helpful to add to `AncillaryDataLib`?

Reply:
well that would mean that the `Governor.sol` also needs to import `AncillaryDataLib` and these particular methods are only used in the `Governor`.

Reply:
Yeah, I was trying to keep the changes as minimal as possible. We could move them into ancillary data lib, but they are mostly unrelated to ancillary data since they are used to construct the identifier. Note: there is some functionality in the ancillary data lib that does a similar thing (more efficiently), but I wanted to be super confident that this works _exactly_ the same way the governor does.

Reply:
Agreed with both of you, nothing to do!